### ISSUE TITLE
Bike Routing not using 3d information = always using shortest route

### ISSUE THREAD

**Issue by aerostitch** [aerostitch/testnavit#39]
*Issue migrated from trac ticket # 641*

**component:** core | **priority:** minor | **keywords:** srtm, heights

#### 2010-08-24 06:52:45: robin created the issue

___
> If I change to vehicle = bike the routing with osm maps will always guide me to pass over a hill rather than to cycle around, which timewise is much shorter since the hill is very steep.
> I checked with http://openrouteservice.org/ and there the layout "racing bike" (equivalent to Rennrad) returns a valid result also the "savest way"(sicherster Weg) and "cycle route" work fine.
> Is there a simple way to adjust this in the xml file? and where does navit get its height coordinate from, as to my understanding osm has only 2d data?

**Comment by aerostitch** [comment]
#### 2010-08-24 10:48:59: korrosa changed priority from *major* to *minor*

**Comment by aerostitch** [comment]
#### 2010-08-24 10:48:59: korrosa changed type from *defect/bug* to *enhancement/feature request*

**Comment by aerostitch** [comment]
#### 2010-08-24 10:48:59: korrosa commented


___
> OSM only has 2D data. Navit cannot take into account any fluctuations in height when routing using OSM maps. The only difference between the routes created by vehicle:bike and vehicle:car (in the default navit.xml) are that the bike routes can follow cycle paths, whilst the car cannot.
> 
> I don't know whether using other maps which do have 3D data (such as Garmin maps) enable Navit to take into account height fluctuations. If it does, then you're onto a winner because many people join OSM and NASA data to make 3D Garmin maps of a variety of countries. (The NASA data is basically topography of the planet, if I'm not mistaken: http://www2.jpl.nasa.gov/srtm/ and http://en.wikipedia.org/wiki/Shuttle_Radar_Topography_Mission)
> 
> (Changed type of ticket, as it's not quite a bug. Unless Navit says somewhere that it can do 3D routing using OSM which I haven't seen...)

**Comment by aerostitch** [comment]
#### 2010-11-05 09:21:01: @pgrandin

**Comment by aerostitch** [comment]
#### 2010-11-05 09:21:01: @pgrandin commented


___
> The idea is interesting, but indeed currently navit doesnt do 3d routing.
> 
> Patch welcome!

**Comment by aerostitch** [comment]
#### 2010-11-09 02:26:36: @pgrandin commented


___
> #649 has been marked as a duplicate of this ticket.

**Comment by aerostitch** [comment]
#### 2012-09-02 17:47:10: chaoscrawler@gmail.com commented


___
> i've thought about dirty hack to simulate '3d routing' by exploiting already implemented maxspeed tag, or by using the surface/road type tag.
> 
> idea of hack is to create custom binfile , based on OSM and 3d data, which would create extra ways in parallel to current ones - each oneway - on the hills (uphill would be slower and downhill - faster) . original road has to be removed or tagged with slowest speed of the two oneway roads.
> 
> idea is to tweak uphill and downhill maxspeed . ofcourse such data would be useless for cars (except they could perhaps save fuel by using such data) .
> 
> another side effect would be that result bifile would grow in size and routing would be slower because more data would have to be parsed - but imho for bicycle routing, where avg. routes are much shorter this is quite acceptable - afterall result will contain at least a bit more 3d data (not enough to display terrain, but enough for routing decisions)
>  
> 
> atm i have no patch to implement this in practice - just the rough idea. 
>

**Comment by aerostitch** [comment]
#### 2013-07-03 01:51:29: usul commented


___
> This feature request has a bit wider scope than just bicycling. Even for **eMobility**, avoiding rides onto mountains (or make use of them to save energy) is essential.
> 
> I heared about SRTM (public 3D dataset) processing for Navit at the IRC, but I'm not sure if this was routing focused.
> 
> Please keep in mind, that this issue has **high complexity** and so might not beeing realized in the next releases.
